File: app/src/main/java/dev/olog/basil/presentation/main/di/MainActivityModule.java
package dev.olog.basil.presentation.main.di;

import android.content.Context;

import androidx.appcompat.app.AppCompatActivity;
import dagger.Module;
import dagger.Provides;
import dev.olog.basil.dagger.qualifier.ActivityContext;
import dev.olog.basil.presentation.main.MainActivity;

@Module
public class MainActivityModule {

    private final MainActivity activity;

    MainActivityModule(MainActivity activity) {
        this.activity = activity;
    }

    @Provides
    @ActivityContext
    Context provideContext(){
        return activity;
    }

    @Provides
    AppCompatActivity provideActivity(){
        return activity;
    }

}

File: presentation/src/main/java/dev/olog/basil/presentation/main/di/MainActivityModule.kt
package dev.olog.basil.presentation.main.di

import androidx.lifecycle.ViewModel
import dagger.Binds
import dagger.Module
import dagger.multibindings.IntoMap
import dev.olog.basil.presentation.ViewModelKey
import dev.olog.basil.presentation.main.RecipesViewModel

@Module
abstract class MainActivityModule {

    @Binds
    @IntoMap
    @ViewModelKey(RecipesViewModel::class)
    internal abstract fun provideAlbumViewModel(viewModel: RecipesViewModel): ViewModel

}